pkg/assembler/backends/neo4j/backend.go

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"context"
+
+	"github.com/guacsec/guac/pkg/assembler/backends"
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+)
+
+type Neo4jCredentials struct {
+	User   string
+	Pass   string
+	Realm  string
+	DBAddr string
+}
+
+type neo4jClient struct {
+	driver neo4j.Driver
```

Comment:
Do we need a `backend.Close()` as part of the interface?

Reply:
I'm not sure, these are singletons for the entire duration of the service. The driver for neo4j is just an opaque struct to contain login info, when we do `driver.Session` it tries to connect again, afaik. and then we do `defer session.Close()` in the backend implementation when we get the session